app/src/reducers/queryReducer.js

```diff
@@ -14,16 +14,7 @@ export default function query(state = initialState.query, action) {
       };
       return newState;
     case SUBMIT_QUERY:
-      newState = {
-        current: state.current,
-        lastSubmitted: state.current,
-        submitted: true,
-        isWaiting: true,
-        results: [], //new array to clear out old results upon new submission
-      };
-      return newState;
-    case FETCH_QUERY:
-      return action;
+      return Object.assign({}, state, { submitted: true, isWaiting: true, results: [] });
```

Comment:
@fzxu, unless you have come across something indicating that use of `Object.assign(...)` is a community preference, I'd rather we use another syntax, which I started using in newly added reducer functions in `entityReducer.js`

```
newState = {
  ...state,
  submitted: true,
  isWaiting: true,
  results: [],
}
return newState;
```

specifically, the `...state` line in the construction of the new object, does exactly what the second param of your `Object.assign(...)` call does. That is, it uses attrs from existing `state` obj and then overwrites any that are explicitly listed. This uses the ES6 expand operator, which can be a bit more elegant

in this case, assigning to newState is a bit redundant before returning it... but there are some other cases where having the temp var is important so I'd like to keep using it for most reducer cases

Reply:
I think maybe the `Object.assign` was the 'old preference'. Yours looks cleaner. I will make the change.